File: smashpen/src/main/java/com/smashpen/smashpen/repository/CharacterRepository.java
package com.smashpen.smashpen.repository;

import com.smashpen.smashpen.domain.SmashCharacter;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CharacterRepository extends JpaRepository<SmashCharacter, Long> {
	SmashCharacter findByName(String name);
}

File: smashpen/src/main/java/com/smashpen/smashpen/repository/CharacterRepository.kt
package com.smashpen.smashpen.repository

import com.smashpen.smashpen.domain.SmashCharacter
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CharacterRepository : JpaRepository<SmashCharacter, Long> {
    fun findByName(name: String?): SmashCharacter
}
